### PR TITLE
Features/unit tests

### PR DESCRIPTION
**Pull request by MohamedBeydoun**
Added unit tests for controllers, config, and util functions

### CODE REVIEW

**Review by aehwany — APPROVED** [review]

File: src/controllers/auth.ts [hunk 1]
```diff
@@ -19,20 +19,20 @@ const authController = {
                 const { email, password } = req.body;
                 const user: IUserModel = await userDBInteractions.findByEmail(email, "+password");
                 if (!user) {
-                    res.status(statusCodes.BAD_REQUEST).send({ status: statusCodes.BAD_REQUEST, message: "Invalid email or password" });
+                    res.status(statusCodes.BAD_REQUEST).json({ status: statusCodes.BAD_REQUEST, message: "Invalid email or password" });
```
Comment (imad-ant) [inline src/controllers/auth.ts:22]: Nice catch

File: tests/unit/controllers/auth.test.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                    email: testUser.email,
+                    password: "password"
+                }
+            });
+        });
+
+        it("status 200: returns new user", async () => {
+            stubs.userValidator.validationResult.returns({ isEmpty() { return true; } });
+            stubs.userDB.findByEmail.returns(testUser);
+            stubs.userUtil.codeforces.updateUserProblems.returns();
```
Comment (imad-ant) [inline tests/unit/controllers/auth.test.ts:122]: This statement is redundant - for dev readability I think `.returns(null)` is preferable, or just omitting it since we should really only stub returns if we want it to return something IMO.

File: tests/unit/controllers/auth.test.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        let req;
+        beforeEach(() => {
+            req = mockReq({
+                query: {
+                    token: testToken
+                }
+            });
+        });
+
+        it("status 200: returns a token with it's user", async () => {
```
Comment (imad-ant) [inline tests/unit/controllers/auth.test.ts:73]: Grammar - `its` instead of `it's` for possessive.

File: tests/unit/controllers/auth.test.ts [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+            stubs.userValidator.validationResult.returns({ isEmpty() { return true; } });
+            await authController.introspect(req, mockRes);
+            sinon.assert.calledWith(mockRes.status, statusCodes.SUCCESS);
+            sinon.assert.calledWith(mockRes.json, { active: true, user: { email: testUser.email, id: `${testUser._id}`, role: testUser.role, iat: testTokenIat } });
+        });
+
+        it("status 422: returns an appropriate response with validation errors", async () => {
```
Comment (imad-ant) [inline tests/unit/controllers/auth.test.ts:80]: Maybe we can rephrase this as "given validation error" instead of "with validation errors"?

**Review by imad-ant — COMMENTED** [review]
For next set of testing changes
